clai/clai/web/cli.py

```diff
@@ -0,0 +1,141 @@
+"""CLI command for launching a web chat UI for discovered agents."""
+
+from __future__ import annotations
+
+import logging
+
+from pydantic import ValidationError
+from rich.console import Console
+
+from pydantic_ai import Agent
+from pydantic_ai._cli import _load_agent  # pyright: ignore[reportPrivateUsage]
+from pydantic_ai.builtin_tools import (
+    BUILTIN_TOOL_CLASSES,
+    BUILTIN_TOOL_ID,
+    AbstractBuiltinTool,
+)
+from pydantic_ai.exceptions import UserError
+from pydantic_ai.models import infer_model
+from pydantic_ai.ui.web import create_web_app, load_mcp_server_tools
+
+
+def run_web_command(  # noqa: C901
+    agent_path: str | None = None,
+    host: str = '127.0.0.1',
+    port: int = 7932,
+    models: list[str] | None = None,
+    tools: list[str] | None = None,
+    instructions: str | None = None,
+    mcp: str | None = None,
+) -> int:
+    """Run the web command to serve an agent via web UI.
+
+    Args:
+        agent_path: Agent path in 'module:variable' format. If None, creates generic agent.
+        host: Host to bind the server to
+        port: Port to bind the server to
+        models: List of model strings (e.g., ['openai:gpt-5', 'claude-sonnet-4-5'])
+        tools: List of builtin tool IDs (e.g., ['web_search', 'code_execution'])
+        instructions: System instructions for generic agent
+        mcp: Path to JSON file with MCP server configurations
+    """
+    console = Console()
+
+    if agent_path:
+        agent = _load_agent(agent_path)
+        if agent is None:
+            console.print(f'[red]Error: Could not load agent from {agent_path}[/red]')
+            return 1
+    else:
+        agent = Agent()
+
+        if instructions:
+
+            @agent.system_prompt
+            def system_prompt() -> str:  # pyright: ignore[reportUnusedFunction]
+                return instructions
+
+    if agent.model is None and not models:
+        console.print('[red]Error: At least one model (-m) is required when agent has no model[/red]')
+        return 1
+
+    parsed_tools_ids: list[BUILTIN_TOOL_ID] = []
+    if tools:
+        for tool_id in tools:
+            if tool_id not in BUILTIN_TOOL_ID.__args__:
+                logging.warning(
+                    f'Tool "{tool_id}" is not a valid builtin tool ID. Valid IDs are: {BUILTIN_TOOL_ID.__args__}'
+                )
+                continue
+            parsed_tools_ids.append(tool_id)  # pyright: ignore[reportArgumentType]
+
+    parsed_model_ids: list[str] = []
+
+    if models:
+        for model_str in models:
+            if ':' in model_str:
```

Comment:
I don't think we need to check this as we end up calling `infer_model` later anyway, so it's ok if `parsed_model_ids` contains a prefixless name